keras_cv/layers/preprocessing/cut_mix.py

```diff
@@ -0,0 +1,142 @@
+
+    Sample usage:
+    ```python
+    (x_train, y_train), (x_test, y_test) = keras.datasets.cifar10.load_data()
+    cutmix = CutMix(10)
+    augmented_data, updated_labels = cutmix(x_train, y_train)
+    ```
+    """
+    def __init__(
+        self, num_classes, alpha=0.8, probability=1.0, label_smoothing=0.0, **kwargs
+    ):
+        super(CutMix, self).__init__(*kwargs)
+        self.alpha = alpha
+        self.probability = probability
+        self.num_classes = num_classes
+        self.label_smoothing = label_smoothing
+
+    @staticmethod
+    def _sample_from_beta(alpha, beta, shape):
+        sample_alpha = tf.random.gamma(shape, 1.0, beta=alpha)
+        sample_beta = tf.random.gamma(shape, 1.0, beta=beta)
+        return sample_alpha / (sample_alpha + sample_beta)
+
+    def call(self, images, labels):
+        augment_cond = tf.less(
+            tf.random.uniform(shape=[], minval=0.0, maxval=1.0), self.probability
+        )
+        # pylint: disable=g-long-lambda
+        cutmix_augment = lambda: self._update_labels(*self._cutmix(images, labels))
+        no_augment = lambda: (images, self._smooth_labels(labels))
+        return tf.cond(augment_cond, cutmix_augment, no_augment)
+
+    def _cutmix(self, images, labels):
+        """Apply cutmix."""
+        lambda_sample = CutMix._sample_from_beta(
+            self.alpha, self.alpha, labels.shape
+        )
+
+        ratio = tf.math.sqrt(1 - lambda_sample)
+
+        batch_size = tf.shape(images)[0]
+        image_height, image_width = tf.shape(images)[1], tf.shape(images)[2]
+
+        cut_height = tf.cast(
+            ratio * tf.cast(image_height, dtype=tf.float32), dtype=tf.int32
+        )
+        cut_width = tf.cast(
+            ratio * tf.cast(image_height, dtype=tf.float32), dtype=tf.int32
+        )
+
+        random_center_height = tf.random.uniform(
+            shape=[batch_size], minval=0, maxval=image_height, dtype=tf.int32
+        )
+        random_center_width = tf.random.uniform(
+            shape=[batch_size], minval=0, maxval=image_width, dtype=tf.int32
+        )
+
+        bbox_area = cut_height * cut_width
+        lambda_sample = 1.0 - bbox_area / (image_height * image_width)
+        lambda_sample = tf.cast(lambda_sample, dtype=tf.float32)
+
+        images = tf.map_fn(
+            lambda x: _fill_rectangle(*x),
+            (
+                images,
+                random_center_width,
+                random_center_height,
+                cut_width // 2,
+                cut_height // 2,
+                tf.reverse(images, [0]),
+            ),
+            dtype=(tf.float32, tf.int32, tf.int32, tf.int32, tf.int32, tf.float32),
+            fn_output_signature=tf.TensorSpec(images.shape[1:], dtype=tf.float32),
+        )
+
+        return images, labels, lambda_sample
+
+    def _update_labels(self, images, labels, lambda_sample):
+        labels_1 = self._smooth_labels(labels)
+        labels_2 = tf.reverse(labels_1, [0])
```

Comment:
same question for this one. Probably worth some comment.

Reply:
I renamed the variable, hopefully this clarifies the issue.